week-2/Homework/mandatory/4-radio-stations.js

```diff
@@ -0,0 +1,79 @@
+/**
+ * Finding a radio station, and a good one, can be hard manually.
+ * Let's use some code to help us build a program that helps us scan
+ * the radio waves for some good music.
+ */
+
+/**
+ * First, let's create a function that creates a list of all the frequencies.
+ * Call this function `getAllFrequencies`.
+ *
+ * This function should:
+ * - Create an array  starting at 87 and ending in 108
+ * - Should return this array to use in other functions
+ */
+
+// `getAllFrequencies` goes here
+
+/**
+ * Next, let's write a function that gives us only the frequencies that are radio stations.
+ * Call this function `getStations`.
+ *
+ * This function should:
+ * - Get the available frequencies from `getAllFrequencies`
+ * - There is a helper function called isRadioFrequency that takes an integer as an argument and returns a boolean.
+ * - Sort the stations by low - high e.g. 1,2,3,4,5
+ * - Return only the frequencies that are radio stations.
+ */
+// `getStations` goes here
+
+/**
+ * Lastly, let's make a function for people to use.
+ * When this runs, it should use the above two functions to log to the `console`
+ * Call this function `searchRadioWaves`.
+ * - "🎶 station found at ${frequency}, enjoy!"
+ * - "No station found at ${frequency}, moving on"
+ */
+// `searchRadioWaves` goes here
+
+/* ======= TESTS - DO NOT MODIFY ===== */
+
+function isRadioStation(frequency) {
+  // This is just a way of caching the random array internally
```

Comment:
I'd probably use a different word to cache for now (or start with a simple word and use "cache" in brackets). In this case I'd probably avoid using cache since that's not what it's actually doing (i.e. copying data closer to where it can be retrieved - it's actually being generated here for the first time).Suggested wording: "This is a way of storing the random frequency array inside a function as storing it outside is bad practice".

I'd also add more of an explanation of what this function does to set an example on how to comment code to the students.